### PR TITLE
Mock variableService getMacros 

### PR DESCRIPTION
**Pull request by micajuine-ho**
Mocked `getMacros()`, so that amp-analytics variables get bound to a name transformation, instead of their resolving function:
```
$IF(blah, blah, blah) =>  _IF(blah%2Cblah%2Cblah)_      **notice that commas get encoded**
$NOT(true) => _NOT(true)_
COOKIE(sp_id) => __COOKIE(sp_id)__
$HASH($NOT(true)) => _HASH(_NOT(true)_)_
```

Only 3 vendors in `vendor-requests.js` needed to be changed.

**NOTE** macros in `variables.js` that do not have a `$` preceding them will get expanded twice, which results in double underscores: `${consentState} => __CONSENT_STATE__`. This is due to the nested macros behavior that was recently introduced #26271 where expansion happens twice (and since CONSENT_STATE is still present in the string, it gets expanded again).

### CODE REVIEW

File: extensions/amp-analytics/0.1/test/test-vendors-new.js [hunk 1]
```diff
@@ -104,6 +104,30 @@ describes.realWin(
...
             });
+
+            // Have to get service after analytics element is created
+            const variableService = variableServiceForDoc(doc);
+
+            window.sandbox
+              .stub(variableService, 'getMacros')
+              .callsFake(function() {
+                // Add all the macros in amp-analytics
+                const elementMacros = {
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-vendors-new.js]: Can you store the elementMacros to `this.elementMacros_` maybe? Then we don't need to worry about updating the list here.

File: extensions/amp-analytics/0.1/test/vendor-requests.json [hunk 2]
```diff
@@ -73,7 +73,7 @@
   "comscore": {
     "host": "https://sb.scorecardresearch.com",
     "base": "https://sb.scorecardresearch.com/b?",
-    "pageview": "https://sb.scorecardresearch.com/b?c1=2&c2=1000001&cs_ucfr=&cs_amp_consent=&cs_pv=_page_view_id_&c12=_client_id_&rn=_random_&c8=_title_&c7=_canonical_url_&c9=_document_referrer_&cs_c7amp=_ampdoc_url_"
+    "pageview": "https://sb.scorecardresearch.com/b?c1=2&c2=1000001&cs_ucfr=_IF__IF__IF_&cs_amp_consent=_CONSENT_STATE_&cs_pv=_page_view_id_&c12=_client_id_&rn=_random_&c8=_title_&c7=_canonical_url_&c9=_document_referrer_&cs_c7amp=_ampdoc_url_"
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/vendor-requests.json]: Just realize that the change here would make it difficult to revert change if we want to unlaunch the json vendor config change. 
One option is to change the test-vendor.js as well, another option is to wait for another 2 weeks. What do you think?

**Comment by micajuine-ho** [comment]
Added the mock getMacros to `test-vendors.js` for backward compatibility's sake. 

Ran those tests to verify that that changes were not breaking. Discovered some [incongruences between some json vs js files](), that have to be fixed before this is applied.

File: extensions/amp-analytics/0.1/test/test-vendors-new.js [hunk 3]
```diff
@@ -104,6 +109,34 @@ describes.realWin(
...
+                // Add all the macros in amp-analytics
+                const merged = {...this.macros_, ...elementMacros};
+
+                // Change the resolving function
+                const keys = Object.keys(merged);
+                for (let i = 0; i < keys.length; i++) {
+                  const key = keys[i];
+                  // Three opt params here because that is currently
+                  // the max number of params for any macro
+                  merged[key] = (opt_param, opt_param2, opt_param3) => {
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-vendors-new.js]: can we use `(...args) =>` to support any number of arguments?

File: extensions/amp-analytics/0.1/test/test-vendors-new.js [hunk 4]
```diff
@@ -104,6 +109,34 @@ describes.realWin(
...
+                    let params = [opt_param, opt_param2, opt_param3]
+                      .filter(val => val !== undefined)
+                      .join(',');
+                    if (params) {
+                      params = '(' + params + ')';
+                    }
+                    return `_${key.replace('$', '')}${params}_`;
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-vendors-new.js]: To confirm: 
analytics macros like `EXAMPLE(FAKE, param1, param2)` is replaced to `_EXAMPLE(_fake_, param1,param2)_`
global macros like `CLIENT_ID(param)` is replaced with `_client_id_`

To get the same behavior across all variables, can we please
1. Resolve convert analytics macro to lower case as well. 
2. Include the param value for global variables like CLIENT_ID. 

Feel free to address 2 in a separate PR. Thanks.

File: extensions/amp-analytics/0.1/test/test-vendors-new.js (hunk 3 above)
Comment (micajuine-ho) [reply extensions/amp-analytics/0.1/test/test-vendors-new.js]: Yup, good call :)

File: extensions/amp-analytics/0.1/test/test-vendors-new.js (hunk 4 above)
Comment (micajuine-ho) [reply extensions/amp-analytics/0.1/test/test-vendors-new.js]: Added 1 (which solved the double expansion problem actually!)
Will address 2 in a separate PR as part of this project.

**Pull request merged**